lib/internal/process/next_tick.js

```diff
@@ -5,6 +5,7 @@ exports.setup = setupNextTick;
 function setupNextTick() {
   const promises = require('internal/process/promises');
   const emitPendingUnhandledRejections = promises.setup(scheduleMicrotasks);
+  const MAX_TICKS = 1e4;
```

Comment:
@Fishrock123 that interpretation is incorrect. it guards against the `nextTickQueue` array becoming too large and running the process out of memory. There's a simple test for this which shows the event loop not being allowed to proceed. This is by design, as the alternative was a bit of a pain to work around.
```js
setImmediate(() => process._rawDebug('hi'));
(function runner() { process.nextTick(runner) })();
```